examples/fluids/qfunctions/utils.h

```diff
@@ -40,6 +151,24 @@ CEED_QFUNCTION_HELPER void KMUnpack(const CeedScalar v[6], CeedScalar A[3][3]) {
   A[1][0] = A[0][1] = weight * v[5];
 }
 
+// @brief Pack full tensor into Kelvin-Mandel notation symmetric tensor
+CEED_QFUNCTION_HELPER void KMPack(const CeedScalar A[3][3], CeedScalar v[6]) {
+  const CeedScalar weight = sqrt(2.);
+  v[0]                    = A[0][0];
+  v[1]                    = A[1][1];
+  v[2]                    = A[2][2];
+  v[3]                    = A[2][1] * weight;
+  v[4]                    = A[2][0] * weight;
+  v[5]                    = A[1][0] * weight;
+}
+
+// @brief Calculate metric tensor from mapping, g_{ij} = xi_{k,i} xi_{k,j} = dXdx^T dXdx
+CEED_QFUNCTION_HELPER void KMMetricTensor(const CeedScalar dXdx[3][3], CeedScalar km_g_ij[6]) {
+  CeedScalar g_ij[3][3] = {{0.}};
+  MatMat3(dXdx, dXdx, CEED_TRANSPOSE, CEED_NOTRANSPOSE, g_ij);
+  KMPack(g_ij, km_g_ij);
+}
```

Comment:
Is this a `KM` (Kelvin-Mandel form of a symmetric tensor)?

Reply:
Yep.